plugins/modules/oneview_server_profile.py

```diff
@@ -0,0 +1,674 @@
+#!/usr/bin/python
+###
+# Copyright (2016-2020) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: oneview_server_profile
+short_description: Manage OneView Server Profile resources
+description:
+    - Manage the servers lifecycle with OneView Server Profiles. On C(present) state, it selects a server hardware
+      automatically based on the server profile configuration if no server hardware was provided.
+version_added: "2.5.0"
+requirements:
+    - hpeOneView >= 5.0.0
```

Comment:
change the version to 5.4.0 since hpeOneView module is available from 5.4.0

Reply:
done